compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1203,6 +1244,56 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
   final def lub(tps: List[Type]): Type =
     ((defn.NothingType: Type) /: tps)(lub(_,_, canConstrain = false))
 
+  def lubArgs(args1: List[Type], args2: List[Type], tparams: List[TypeParamInfo], canConstrain: Boolean = false): List[Type] =
```

Comment:
Complications: Need added case for lub and glb of applied types, which mimic the simplifications done for refined types.